api/api.go

```diff
@@ -21,5 +22,6 @@ func Setup(ctx context.Context, router *mux.Router, jobStore store.JobStore) *Jo
 
 	router.HandleFunc("/jobs", api.CreateJobHandler(ctx)).Methods("POST")
 	router.HandleFunc("/jobs/{id}", api.GetJobHandler(ctx)).Methods("GET")
+	router.HandleFunc("/jobs", api.GetJobsHandler(ctx)).Methods("GET")
```

Comment:
I can take out the ctx, I understand what you've said, but if I do that then it gives me an error where I have used it in a log.Event just before the handler function is created. I can either remove the logging there or put it inside the handler function itself (where I can use the request context, as you've explained). What would you prefer?

Reply:
If the logging isn't there for a specific purpose it begs the question of if it's really necessary in the first place? Another solution is just to pass context.TODO() to the logger (it doesn't seem like it particularly needs any details from the context).
If you do remove the logging and the context then there's no reason to have a function that returns a function, you can just use the Handlerfunc directly.

Reply:
I've now removed the context and changed it to use the HandlerFunc directly, as suggested by @ChrysmOre. I've changed the logging so that it shows when the handler function is being entered (my previous tech lead taught me that this is good practice).